src/__tests__/linter.jest.test.ts

```diff
@@ -66,6 +70,48 @@ describe('Linter', () => {
     );
   });
 
+  it('should expose function-live lifespan cache to custom functions', async () => {
```

Comment:
@P0lip Could you please add a test showing that the cache is not shared among different functions?